libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -322,14 +322,21 @@ internal class MapboxTripSession(
             val routeCoordinates = tripStatus.route?.routeOptions()?.coordinatesList()
             return if (routeCoordinates != null) {
                 val waypointsCount = routeCoordinates.size
-                // TODO: get rid of the workaround https://github.com/mapbox/mapbox-navigation-android/issues/5275
-                val nextWaypointIndex = max(
-                    MIN_NEXT_WAYPOINT_INDEX,
+                val nextWaypointIndex = normalizeNextWaypointIndex(
                     tripStatus.navigationStatus.nextWaypointIndex
                 )
                 return waypointsCount - nextWaypointIndex
             } else 0
         }
+
+        /**
+         * SDK always navigates from the current location.
+         * The Android SDK acts as a user always reached the origin, even if he doesn't.
```

Comment:
NIT for better understanding. Could you add something around these lines https://github.com/mapbox/mapbox-navigation-android/pull/5327#issue-1094410795? 👇 

> On the Android side we always start navigation from the current position and expect that the next waypoint index will not be less than 1. But native part considers the origin as usual waypoint can return next waypoint index 0.

Reply:
We might want to try and highlight the fact that this is only about reroutes. When we're rerouting, we never want to include the origin of the route, even if the origin has not been reached yet (due to some conditions).

Reply:
updated, thanks